src/transport/raw/TCPConfig.h

```diff
@@ -0,0 +1,144 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines default compile-time configuration constants
+ *      for CHIP.
+ *
+ *      Package integrators that wish to override these values should
+ *      either use preprocessor definitions or create a project-
+ *      specific chipProjectConfig.h header and then assert
+ *      HAVE_CHIPPROJECTCONFIG_H via the package configuration tool
+ *      via --with-chip-project-includes=DIR where DIR is the
+ *      directory that contains the header.
+ *
+ *
+ */
+
+#pragma once
+
+#include <lib/core/CHIPConfig.h>
+
+namespace chip {
+
+/**
+ * @def CHIP_CONFIG_CONTROLLER_MAX_ACTIVE_TCP_CONNECTIONS
+ *
+ * @brief Maximum Number of TCP connections a controller can simultaneously have
+ */
+#ifndef CHIP_CONFIG_CONTROLLER_MAX_ACTIVE_TCP_CONNECTIONS
+#define CHIP_CONFIG_CONTROLLER_MAX_ACTIVE_TCP_CONNECTIONS 4
+#endif
+
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT && CHIP_CONFIG_CONTROLLER_MAX_ACTIVE_TCP_CONNECTIONS < 1
+#error "If TCP is enabled, the controller needs to support at least 1 TCP connection"
+#endif
+
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT && CHIP_CONFIG_CONTROLLER_MAX_ACTIVE_TCP_CONNECTIONS > INET_CONFIG_NUM_TCP_ENDPOINTS
+#error "If TCP is enabled, the maximum number of connections cannot exceed the number of tcp endpoints"
+#endif
+
+/**
+ * @def CHIP_CONFIG_DEVICE_MAX_ACTIVE_TCP_CONNECTIONS
+ *
+ * @brief Maximum Number of TCP connections a device can simultaneously have
```

Comment:
Why is this a separate ifdef from "controller"?

What are the definitions of "device" and "controller" here for a single node that would use both ifdefs?  I have no idea how I would set these values for a node I am creating.

Reply:
Collapsed into one ifdef which should be settable by both device and controller at their respective ends.